oauth.js

```diff
@@ -0,0 +1,89 @@
+var crypto = require('crypto');
+var merge = require('deepmerge');
+
+/**
+ * Create oauth compliant basestring, signing key, and then signature
+ * @param {String} method
+ * @param {String} url
+ * @param {String} apiSecret
+ * @param {String} tokenSecret
+ * @param {object} params
+ * @returns {String}
+ */
+function createSignature(method, url, apiSecret, tokenSecret, params) {
+
+	var baseString = method.toUpperCase() + '&' + encodeURIComponent(url) + '&',
+	    signingKey = apiSecret + "&" + (tokenSecret || '');
+
+	Object.keys(params).sort().forEach(function (key, i) {
```

Comment:
Tried replacing this with querystring, but it doesn't solve the sorted order in a good reliable way.  This works and matches what the API expects.